src/components/MapView/DateSelector/index.tsx

```diff
@@ -31,41 +36,109 @@ const Input = forwardRef(
   },
 );
 
-function DateSelector({ unavailableDates = [], classes }: DateSelectorProps) {
+function DateSelector({ availableDates = [], classes }: DateSelectorProps) {
   const selectedDate = useSelector(selectCurrentDate);
+  const selectedMonth = selectedDate.getMonth();
+  const selectedYear = selectedDate.getFullYear();
   const dispatch = useDispatch();
 
-  function isUnavailableDate(date: Date) {
+  function isAvailableDate(date: Date) {
     return Boolean(
-      unavailableDates.find(dateIt => moment(date).isSame(dateIt, 'day')),
+      availableDates.find(dateIt => moment(date).isSame(dateIt, 'day')),
     );
   }
 
+  function isAvailableMonth(month: number) {
+    return Boolean(
+      findAvailableDayInMonth(new Date(selectedYear, month, 1), availableDates),
+    );
+  }
+
+  function updateMonth(month: number) {
+    // Fetching the first available day in the month
+    const date = findAvailableDayInMonth(
+      new Date(selectedYear, month, 1),
+      availableDates,
+    );
+    if (date) {
+      dispatch(updateDate(date));
+    }
+  }
+
+  function incrementYear() {
+    dispatch(updateDate(new Date(selectedYear + 1, selectedMonth, 1)));
+  }
+
+  function decrementYear() {
+    dispatch(updateDate(new Date(selectedYear - 1, selectedMonth, 1)));
```

Comment:
So the way this works right here, when we incr/decr the year on the timeline, it changes both the timeline element AND the selected date for data display purposes, right? (and potentially sets the "active" date to a date with no data available, since it defaults to the first of the month in a different year). Am I understanding that correctly?

I think we probably want to separate that state out into 2 different pieces:

- 1 date that is the "active" date for purposes of displaying data on the map. This is the date that gets set when you click on a month (or on a day in the calendar picker).

- The current year that the timeline is showing. This lets you quickly scroll through the timeline by year to see when data is available for without the map constantly changing/updating (which might involve a lot of data downloads, or putting us into dates with no data available).